ament_cmake_export_interfaces/cmake/ament_cmake_export_interfaces-extras.cmake.in

```diff
@@ -5,7 +5,24 @@ set(_exported_interfaces "@_AMENT_CMAKE_EXPORT_INTERFACES@")
 # include all exported interfaces
 if(NOT _exported_interfaces STREQUAL "")
   foreach(_interface ${_exported_interfaces})
-    include("${@PROJECT_NAME@_DIR}/${_interface}Export.cmake")
-    list(APPEND @PROJECT_NAME@_INTERFACES "@PROJECT_NAME@::${_interface}")
+    set(_export_file "${@PROJECT_NAME@_DIR}/${_interface}Export.cmake")
```

Comment:
@dirk-thomas just to make sure I understand, we're going with this patch to avoid having to change the `ament_export_interfaces()` API and all its use cases (i.e. so that what we export and what we expect when we use `${..._INTERFACES}` match), correct?

Reply:
With changing the API you mean requiring the user to pass the target names explicitly (which must match the targets listed in the corresponding `install()` call)? Yes, that would be an alternative.

I didn't consider it since the native CMake API only requires the targets to be listed in the `install(TARGETS)` call where as the `install(EXPORT)` call only needs the export name and doesn't require the target names to be repeated. I thought it makes more sense to match the API from CMake (as well as avoid that the user has to repeat the same information in the ament API).

Reply:
That's reasonable.